core/src/test/java/org/infinispan/test/transport/DelayedViewJGroupsTransport.java

```diff
@@ -19,11 +24,21 @@ public void receiveClusterView(View newView) {
       // check if this is an event of node going down, and if so wait for a signal to apply new view
       if (waitLatch != null && getMembers().size() > newView.getMembers().size()) {
          try {
-            waitLatch.await(10, TimeUnit.SECONDS);
+            log.debugf("Delaying view %s", newView);
+            unblocked = waitLatch.await(10, TimeUnit.SECONDS);
+            if (unblocked) {
+               log.debugf("Unblocking view %s", newView);
+            } else {
+               log.errorf("Timed out waiting for view to be unblocked: %s", newView);
```

Comment:
Should this throw an exception instead?

Reply:
JGroups may or may not log the exception, but the test is not going to see it anyway.